lib/dialogs/DialogManager.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+
+import { AnimatePresence } from 'framer-motion';
+import Dialog from '~/components/ui/Dialog';
+import { type Dialog as DialogType } from './dialog-schemas';
+import { useDialogStore } from './dialog-store-provider';
+
+const DialogManager = () => {
+  const { closeDialog, dialogs } = useDialogStore((state) => state);
+
+  const handleOpenChange = (dialog: DialogType) => {
+    if (dialog.type === 'Confirm' && dialog.onCancel) {
+      dialog.onCancel();
+      closeDialog(dialog.id);
+      return;
+    }
+    confirmDialog(dialog);
+  };
+
+  const confirmDialog = (dialog: DialogType) => {
+    const result = dialog?.onConfirm?.();
+    // close the dialog if it's not a prompt dialog or the prompt dialog returns a truthy value
+    if (dialog?.type !== 'Prompt' || result) {
+      closeDialog(dialog.id);
+    }
+  };
+
+  const cancelDialog = (dialog: DialogType) => {
+    if (dialog?.type === 'Confirm') {
+      dialog.onCancel?.();
+    }
+    closeDialog(dialog.id);
+  };
+
+  return (
+    <AnimatePresence>
```

Comment:
The idea with me putting animate presence here was that it could be used for exit animations. It seems to be unused currently.